lib/Dialect/FIRRTL/Export/FIREmitter.cpp

```diff
@@ -803,6 +821,33 @@ void Emitter::emitStatement(InstanceOp op) {
   }
 }
 
+void Emitter::emitStatement(InstanceChoiceOp op) {
+  startStatement();
+  auto legalName = legalize(op.getNameAttr());
+  ps << "instchoice " << PPExtString(legalName) << " of "
+     << PPExtString(legalize(op.getDefaultTargetAttr().getAttr())) << ", "
+     << PPExtString(legalize(op.getOptionNameAttr())) << " :";
+  emitLocation(op);
+  ps.scopedBox(PP::bbox2, [&] {
+    for (const auto [optSym, targetSym] : op.getTargetChoices()) {
+      ps << PP::newline;
+      ps << PPExtString(legalize(optSym.getLeafReference()));
+      ps << " => ";
+      ps << PPExtString(legalize(targetSym.getAttr()));
+    }
+  });
+  setPendingNewline();
+
+  SmallString<16> portName(legalName);
+  portName.push_back('.');
+  unsigned baseLen = portName.size();
+  for (unsigned i = 0, e = op.getNumResults(); i < e; ++i) {
+    portName.append(legalize(op.getPortName(i)));
+    addValueName(op.getResult(i), portName);
+    portName.resize(baseLen);
+  }
+}
```

Comment:
This looks good. Thanks for remembering to pass this through legalization. This should enable numeric identifiers here which is, admittedly odd, but consistent with everything else.